File: 백준/Bronze/8958. OX퀴즈/OX퀴즈.py
import sys

n = int(sys.stdin.readline())
test_cases = []

for i in range(n):
    test_cases.append(sys.stdin.readline().strip())

for test_case in test_cases:
    score = 0
    test_case = test_case.split("X")
    for i in test_case:
        j = 1
        while j <= len(i):
            score += j
            j += 1

    print(score)

File: 백준/Bronze/8958. OX퀴즈/OX퀴즈.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

        int n = Integer.parseInt(br.readLine());
        String[] testCases = new String[n];

        for (int i = 0; i < n; i++){
            testCases[i] = br.readLine();
        }

        for (int i = 0; i < n; i++){
            String testCase = testCases[i];
            int score = 0;
            int consecutive = 0;

            for (int j = 0; j < testCase.length(); j++){
                if (testCase.charAt(j) == 'O') {
                    consecutive++;
                    score += consecutive;
                } else {
                    consecutive = 0;
                }
            }
            System.out.println(score);
        }

    }
}
